mvp-spec/dsp-implementation.md

```diff
@@ -545,268 +662,26 @@ in the Audit Logs.
                     "adomain": [ "advertiserdomain.com" ],
                     "cid": "campaign111",
                     "crid": "creative112",
-                    "attr": [ 1, 2, 3, 4, 5, 6, 7, 12 ]
+                    "attr": [ 1, 2, 3, 4, 5, 6, 7, 12 ],
+                    "ext": {
+                        "paf": {
+                            "version": 0,
+                            "receiver": "dsp1.com",
+                            "status": "success",
+                            "details": "",
+                            "source": {
+                                "domain": "dsp1.com",
+                                "timestamp": 1639589531,
+                                "signature": "12345_signature"
+                            },
```

Comment:
Hi, Team!

Could you, please, validate if i get right that all in all, transactions placement in open rtb protocol shall look as follows?
I was not able to find expicit examples for openrtb protocol in spec.

Reqeuest to ssp:
```
{
    "id": "80ce30c53c16e6ede735f123ef6e32361bfc7b22",
    "at": 1, 
    "cur": [ "USD" ],
    "imp": [
        {
            "id": "1",
            "bidfloor": 0.03,
            "banner": {
                "h": 250,
                "w": 300,
                "pos": 0
            },
            "ext": {
                "paf": {
                    "version": 0,
                    "seed": {
                        "version": 0,
                        "transaction_id": "a0651946-0f5b-482b-8cfc-eab3644d2743",
                        "publisher": "publisher.com",
                        "source": {
                            "domain": "publisher0.com",
                            "timestamp": 1639589531,
                            "signature": "12345_signature"
                        }
                    },
                    "parents": []
                }
            }
        }
    ],
```
Reqeuest to dsp:
    1) publisher->ssp transaction signed by ssp placed into "parents section"
```
{
    "id": "80ce30c53c16e6ede735f123ef6e32361bfc7b22",
    "at": 1, 
    "cur": [ "USD" ],
    "imp": [
        {
            "id": "1",
            "bidfloor": 0.03,
            "banner": {
                "h": 250,
                "w": 300,
                "pos": 0
            },
            "ext": {
                "paf": {
                    "version": 0,
                    "seed": {
                        "version": 0,
                        "transaction_id": "a0651946-0f5b-482b-8cfc-eab3644d2743",
                        "publisher": "publisher.com",
                        "source": {
                            "domain": "publisher0.com",
                            "timestamp": 1639589531,
                            "signature": "12345_signature"
                        }
                    },
                    "parents": [
                        {
                            "version": 0,
                            "receiver": "ssp1.net",
                            "status": "success",
                            "details": "",
                            "source": {
                                "domain": "ssp1.net",
                                "timestamp": 1639589531,
                                "signature": "12345_signature"
                            },
                        }
                    ]
                }
            }
        }
    ],
```
Response from dsp to ssp:
    1) ssp->dsp transaction signed by dsp
    2) no children
```
{
    "id": "1234567890",
    "bidid": "abc1123",
    "cur": "USD",
    "seatbid": [
        {
            "seat": "512",
            "bid": [
                {
                    "id": "1",
                    "impid": "1",
                    "price": 9.43,
                    "nurl": "http://adserver.com/winnotice?impid=102",
                    "iurl": "http://adserver.com/pathtosampleimage",
                    "adomain": [ "advertiserdomain.com" ],
                    "cid": "campaign111",
                    "crid": "creative112",
                    "attr": [ 1, 2, 3, 4, 5, 6, 7, 12 ],
                    "ext": {
                        "paf": {
                            "version": 0,
                            "receiver": "dsp1.com",
                            "status": "success",
                            "details": "",
                            "source": {
                                "domain": "dsp1.com",
                                "timestamp": 1639589531,
                                "signature": "12345_signature"
                            },
                            "children": []
                        }
                    }
                }
            ]
        }
    ]
}
```
Response from ssp to publisher:

ext.paf has
    1) previousely generated publisher->ssp signature
    2) dsp signature of ssp->dsp transaction in children section:
```
{
    "id": "1234567890",
    "bidid": "abc1123",
    "cur": "USD",
    "seatbid": [
        {
            "seat": "512",
            "bid": [
                {
                    "id": "1",
                    "impid": "1",
                    "price": 9.43,
                    "nurl": "http://adserver.com/winnotice?impid=102",
                    "iurl": "http://adserver.com/pathtosampleimage",
                    "adomain": [ "advertiserdomain.com" ],
                    "cid": "campaign111",
                    "crid": "creative112",
                    "attr": [ 1, 2, 3, 4, 5, 6, 7, 12 ],
                    "ext": {
                        "paf": {
                            "version": 0,
                            "receiver": "ssp1.net",
                            "status": "success",
                            "details": "",
                            "source": {
                                "domain": "ssp1.net",
                                "timestamp": 1639589531,
                                "signature": "12345_signature"
                            },
                            "children": [
                                {
                                    "version": 0,
                                    "receiver": "dsp1.com",
                                    "status": "success",
                                    "details": "",
                                    "source": {
                                        "domain": "dsp1.com",
                                        "timestamp": 1639589531,
                                        "signature": "12345_signature"
                                    },
                                }
                            ]
                        }
                    }
                }
            ]
        }
    ]
}
```

Reply:
Thank you, Vasilii for your question. This is exactly that! 

The docs were designed for providing implementation details by actors. However, the repo misses indeed a document with a general overview of the flow of the transmissions. Considering adding it. :-)